### PR TITLE
Add search block

### LINKED ISSUE #226: FSE - Search block
## Feature description

Search block for those other than the search in the header
See #234 

Could override default WP Core block markup for search, needed for 404

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

1. Support all other search boxes outside of the header one in #234 (potentially the same)
   1. Inside Hamburger #235 
   2. On 404 Page

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by PatelUtkarsh**
## Summary

Fixes #226 

image

## Checklist

- [ ] My pull request is addressing an [open issue](https://github.com/material-components/material-design-for-wordpress/issues) (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/material-components/material-design-for-wordpress/blob/main/contributing/engineering.md#tests).
- [ ] My code follows the [Contributing Guidelines](https://github.com/material-components/material-design-for-wordpress/blob/main/contributing.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: theme/assets/src/block-editor/blocks/search/edit.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
```
Comment (emeaguiar) [inline theme/assets/src/block-editor/blocks/search/edit.js:19]:
```suggestion
```

File: theme/assets/src/block-editor/blocks/search/edit.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { useBlockProps } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Edit.
+ *
+ * @return {JSX.Element} Block edit.
+ */
+const Edit = () => {
+	const search = __( 'Search', 'material-design-google' );
```
Comment (emeaguiar) [inline theme/assets/src/block-editor/blocks/search/edit.js:32]:
```suggestion
	const search = __( 'Search', 'material-design-google' );

```

File: theme/assets/src/block-editor/blocks/search/edit.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	return (
+		<div { ...useBlockProps() }>
```
Comment (emeaguiar) [inline theme/assets/src/block-editor/blocks/search/edit.js:34]: This should be a `form` element.

File: theme/assets/src/block-editor/blocks/search/edit.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+			{ /* eslint-disable-next-line jsx-a11y/label-has-for */ }
+			<label className="mdc-text-field mdc-text-field--outlined mdc-text-field--with-trailing-icon">
+				<input
+					className="mdc-text-field__input"
+					type="text"
+					aria-labelledby={ search }
```
Comment (emeaguiar) [inline theme/assets/src/block-editor/blocks/search/edit.js:40]: `aria-labelledby` is meant to use an ID, either add a unique ID to the `form` element or switch this to `aria-label`

**Review by emeaguiar — CHANGES_REQUESTED** [review]

**Comment by emeaguiar** [comment]
@PatelUtkarsh Just a couple notes in the block I noticed, otherwise looks good 👍

**Comment by PatelUtkarsh** [comment]
@emeaguiar Thank you for the feedback, I've updated code and resolved in f189f66.

**Review by emeaguiar — APPROVED** [review]

**Pull request merged**